File: EcoArt/src/commande/commande.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package commande;

import LineOrder.LineOrder;
import java.sql.Date;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/**
 *
 * @author User
 */
public class commande {
    long id_c;
    String nomC;
    int id_client;
    String adresse;
    LocalDate date ;
    int numTel;
    String email; 
    double total;
    private List<LineOrder> lineOrders;


    public commande(String nomC, int id_client, String adresse, LocalDate date, int numTel, String email, double total) {
        this.nomC = nomC;
        this.id_client = id_client;
        this.adresse = adresse;
        this.date = date;
        this.numTel = numTel;
        this.email = email;
        this.total=total;
    }
   

    public commande(String nomC, int id_client, String adresse, int numTel, String email) {
        this.nomC = nomC;
        this.id_client = id_client;
        this.adresse = adresse;
        this.numTel = numTel;
        this.email = email;
    }



   

    public commande() {
    }

    public List<LineOrder> getLineOrders() {
        return lineOrders;
    }

    public void setLineOrders(List<LineOrder> lineOrders) {
        this.lineOrders = lineOrders;
    }
    

    public long getId_c() {
        return id_c;
    }

    public void setId_c(long id_c) {
        this.id_c = id_c;
    }

    public String getNomC() {
        return nomC;
    }

    public void setNomC(String nomC) {
        this.nomC = nomC;
    }

    public int getId_client() {
        return id_client;
    }

    public void setId_client(int id_client) {
        this.id_client = id_client;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public int getNumTel() {
        return numTel;
    }

    public void setNumTel(int numTel) {
        this.numTel = numTel;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

   

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final commande other = (commande) obj;
        if (this.id_c != other.id_c) {
            return false;
        }
        if (this.id_client != other.id_client) {
            return false;
        }
        if (this.numTel != other.numTel) {
            return false;
        }
        if (!Objects.equals(this.nomC, other.nomC)) {
            return false;
        }
        if (!Objects.equals(this.adresse, other.adresse)) {
            return false;
        }
        if (!Objects.equals(this.email, other.email)) {
            return false;
        }
        if (!Objects.equals(this.date, other.date)) {
            return false;
        }
        return true;
    }

    public commande(long id_c, LocalDate date) {
        this.id_c = id_c;
        this.date = date;
        this.lineOrders=new ArrayList<>();
    }
     public void addLineOrder(LineOrder lineOrder) {
        lineOrders.add(lineOrder);
    }


    
    
}

File: testweb/src/Entity/Commande.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\CommandeRepository;



#[ORM\Entity(repositoryClass:CommandeRepository::class)]
class Commande
{
    
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
 
    private ? int $idC=null;

    
    #[ORM\Column(length:50)]
    private ? string $nomc=null;

   
    #[ORM\Column]
    private ?int $idClient=null;

   
    #[ORM\Column(length:50)]
    private ?string $adresse=null;

   
    #[ORM\Column]
    private ? \DateTime $date=null ;

    
    #[ORM\Column]
    private ?int $numtel=null;

    #[ORM\Column(length:50)]
    private ? string $email=null;

   
    #[ORM\Column(length:50)]
    private ? float $total=null;


    public function getIdC(): ?string
    {
        return $this->idC;
    }

    public function getNomc(): ?string
    {
        return $this->nomc;
    }

    public function setNomc(string $nomc): static
    {
        $this->nomc = $nomc;

        return $this;
    }

    public function getIdClient(): ?int
    {
        return $this->idClient;
    }

    public function setIdClient(int $idClient): static
    {
        $this->idClient = $idClient;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): static
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getNumtel(): ?int
    {
        return $this->numtel;
    }

    public function setNumtel(int $numtel): static
    {
        $this->numtel = $numtel;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): static
    {
        $this->email = $email;

        return $this;
    }

    public function getTotal(): ?float
    {
        return $this->total;
    }

    public function setTotal(float $total): static
    {
        $this->total = $total;

        return $this;
    }


}
